66_HLSLBxDFTests/app_resources/tests.hlsl

```diff
@@ -36,6 +35,18 @@ namespace IMATH = Imath;
 using namespace IMF;
 using namespace IMATH;
 using json = nlohmann::json;
+
+// custom hash for float32_t2
+template<>
+struct std::hash<float32_t2>
+{
+    size_t operator()(const float32_t2& v) const noexcept
+    {
+        size_t v1 = std::hash<float32_t>()(v.x);
+        size_t v2 = std::hash<float32_t>()(v.y);
+        return v1 ^ (v2 << 1);
+    }
+};
```

Comment:
make one in `cpp_compat` under `__HLSL_VERSION` note that we have our own `hash_combine` which is very similar to GLM's https://github.com/g-truc/glm/blob/master/glm/gtx/hash.inl

You can fairly easily make `hash_helper` which you can partial specialize for vectors and matrices of all types and dimensions